Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+                }
+            }
+        }
+
+
+        [SerializeField]
+        [Tooltip("If true, the spatial mesh will be visible while the object is in the placing state.")]
+        private bool spatialMeshVisible = true;
+
+        /// <summary>
+        /// If true, the spatial mesh will be visible while the object is in the placing state.
+        /// </summary>
+        public bool SpatialMeshVisible
+        {
+            get => spatialMeshVisible;
+            set
+            {
+                if (spatialMeshVisible != value)
+                {
+                    spatialMeshVisible = value; 
+                }
+            }
+        }
+
+
+
+        [SerializeField]
+        [Tooltip("")]
+        private SpatialAwarenessMeshDisplayOptions initialSpatialMeshDisplayOption;
+
+        /// <summary>
+        /// Get the spatial mesh visibility on start.  If the the spatial mesh is visible by default then do not toggle the
+        /// spatial mesh visibility while placing the game object.  Spatial mesh visibility refers to if the wire frame material is
+        /// on the spatial mesh or not.
+        /// </summary>
+        public SpatialAwarenessMeshDisplayOptions InitialSpatialMeshDisplayOption
+        {
+            get
+            {
+
+                IReadOnlyList<IMixedRealitySpatialAwarenessMeshObserver> observers = spatialAwarenessSystemAcecess.GetDataProviders<IMixedRealitySpatialAwarenessMeshObserver>();
+
+                foreach (IMixedRealitySpatialAwarenessMeshObserver observer in observers)
+                {
+                    return observer.DisplayOption;
+                }
+
+                // if the spatial awareness is not enabled
+                return SpatialAwarenessMeshDisplayOptions.None;   
+            }
+        }
+
+        private IMixedRealityDataProviderAccess spatialAwarenessSystemAcecess;
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            "remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            "if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will 
+        /// remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal
+        /// if there is a hit."
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set
+            {
+                if (rotateAccordingToSurface != value)
+                {
+                    rotateAccordingToSurface = value;
+                }
+            }
+        }
+
+        private const int IgnoreRaycastLayer = 2;
```

Comment:
Should we be using https://docs.unity3d.com/ScriptReference/Physics.IgnoreRaycastLayer.html instead of hardcoding an arbitrary value?

Reply:
Added `Physics.IgnoreRaycastLayer`

Reply:
It looks `UnityEngine.Physics.IgnorRaycastLayer` is a const = 4
![image](https://user-images.githubusercontent.com/53493796/73976167-95b69f80-48dc-11ea-831e-deb35cdee128.png)

But in the UnityEditor the layer is = 2
![image](https://user-images.githubusercontent.com/53493796/73976058-643dd400-48dc-11ea-927a-55e2a24de3a5.png)

If 2 is not used as the layer number, the issue of the raycast returning a hit on the object it is supposed to place comes up again.

Reply:
Ah, does https://docs.unity3d.com/ScriptReference/LayerMask.NameToLayer.html this one work instead?

Reply:
This worked:
```
IgnoreRaycastLayer = LayerMask.NameToLayer("Ignore Raycast");
```